File: _sources/FireflyCore/Texting/Plain.vb
'==========================================================================
'
'  File:        Plain.vb
'  Location:    Firefly.Texting <Visual Basic .Net>
'  Description: 纯文本格式
'  Version:     2009.10.31.
'  Copyright(C) F.R.C.
'
'==========================================================================

Imports System
Imports System.Collections.Generic
Imports Firefly.TextEncoding

Namespace Texting
    Public NotInheritable Class Plain
        Private Sub New()
        End Sub

        Public Shared Function ReadFile(ByVal Path As String, ByVal Encoding As System.Text.Encoding) As String()
            Using s = Txt.CreateTextReader(Path, Encoding, True)
                Dim l As New List(Of String)
                While Not s.EndOfStream
                    Dim Line As String = s.ReadLine
                    If Line <> "" Then
                        Line = Line.Replace("\n", ChrQ(13) & ChrQ(10))
                        Line = Line.Replace("\x5C", "\")
                    End If
                    l.Add(Line)
                End While
                Return l.ToArray
            End Using
        End Function
        Public Shared Sub WriteFile(ByVal Path As String, ByVal Encoding As System.Text.Encoding, ByVal Value As IEnumerable(Of String))
            Using s = Txt.CreateTextWriter(Path, Encoding, True)
                Dim n = 0
                For Each v In Value
                    If v <> "" Then
                        Dim Line As String = v
                        If Line <> "" Then Line = Line.Replace("\", "\x5C")
                        If Line <> "" Then Line = Line.Replace(ChrQ(13) & ChrQ(10), ChrQ(10)).Replace(ChrQ(10), "\n")
                        s.WriteLine(Line)
                    Else
                        s.WriteLine()
                    End If
                    n += 1
                Next
            End Using
        End Sub
        Public Shared Function ReadFile(ByVal Path As String) As String()
            Return ReadFile(Path, TextEncoding.Default)
        End Function
        Public Shared Sub WriteFile(ByVal Path As String, ByVal Value As IEnumerable(Of String))
            WriteFile(Path, TextEncoding.WritingDefault, Value)
        End Sub
    End Class
End Namespace

File: _sources/FireflyCore/Texting/Plain.cs
// ==========================================================================
// 
// File:        Plain.vb
// Location:    Firefly.Texting <Visual Basic .Net>
// Description: 纯文本格式
// Version:     2009.10.31.
// Copyright(C) F.R.C.
// 
// ==========================================================================

using System.Collections.Generic;
using Firefly.TextEncoding;

namespace Firefly.Texting
{
    public sealed class Plain
    {
        private Plain()
        {
        }

        public static string[] ReadFile(string Path, System.Text.Encoding Encoding)
        {
            using (var s = Txt.CreateTextReader(Path, Encoding, true))
            {
                var l = new List<string>();
                while (!s.EndOfStream)
                {
                    string Line = s.ReadLine();
                    if (!string.IsNullOrEmpty(Line))
                    {
                        Line = Line.Replace(@"\n", String32.ChrQ(13) + String32.ChrQ(10));
                        Line = Line.Replace(@"\x5C", @"\");
                    }
                    l.Add(Line);
                }
                return l.ToArray();
            }
        }
        public static void WriteFile(string Path, System.Text.Encoding Encoding, IEnumerable<string> Value)
        {
            using (var s = Txt.CreateTextWriter(Path, Encoding, true))
            {
                int n = 0;
                foreach (var v in Value)
                {
                    if (!string.IsNullOrEmpty(v))
                    {
                        string Line = v;
                        if (!string.IsNullOrEmpty(Line))
                            Line = Line.Replace(@"\", @"\x5C");
                        if (!string.IsNullOrEmpty(Line))
                            Line = Line.Replace(String32.ChrQ(13) + String32.ChrQ(10), String32.ChrQ(10)).Replace(String32.ChrQ(10), @"\n");
                        s.WriteLine(Line);
                    }
                    else
                    {
                        s.WriteLine();
                    }
                    n += 1;
                }
            }
        }
        public static string[] ReadFile(string Path)
        {
            return ReadFile(Path, TextEncoding.TextEncoding.Default);
        }
        public static void WriteFile(string Path, IEnumerable<string> Value)
        {
            WriteFile(Path, TextEncoding.TextEncoding.WritingDefault, Value);
        }
    }
}